### PR TITLE
Optimize/verify auth ticket function

### PR DESCRIPTION
**Pull request by lpoli**
This PR contains the following changes:
1. Optimize verify auth-ticket function
2. Fix sql mock issue

### CODE REVIEW

File: code/go/0chain.net/blobbercore/handler/storage_handler.go [hunk 1]
```diff
@@ -66,41 +65,33 @@ func (fsh *StorageHandler) convertGormError(err error) error {
...
+		return nil, err
 	}
+
 	if refRequested.LookupHash != authToken.FilePathHash {
 		authTokenRef, err := reference.GetReferenceFromLookupHash(ctx, authToken.AllocationID, authToken.FilePathHash)
 		if err != nil {
-			return false, err
-		}
-
-		parentPath := refRequested.ParentPath
```
Comment (moldis) [inline code/go/0chain.net/blobbercore/handler/storage_handler.go:88]: @lpoli There was the bug, when I shared the root folder I was not able to access inner files/folders with authTicket. This code resolved it, not sure why it's commented now

**Review by moldis — CHANGES_REQUESTED** [review]
Questions

**Review by moldis — APPROVED** [review]
Looks good

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**